libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -333,131 +336,160 @@ public Optional<IdentityClaim> findIdentityClaim(List<VcStoreItem> vcStoreItems)
-                                                LOG_CRI_ISSUER.getFieldName(),
-                                                item.getCredentialIssuer());
-                        LOGGER.warn(mapMessage);
-
-                        return Optional.empty();
-                    }
-
-                    return Optional.of(drivingPermitNode);
-                } catch (ParseException | JsonProcessingException e) {
-                    LOGGER.error(
-                            "Error while parsing Driving Permit CRI credential: '{}'",
-                            e.getMessage());
-                    throw new HttpResponseExceptionWithErrorBody(
-                            HttpStatus.SC_INTERNAL_SERVER_ERROR,
-                            ErrorResponse.FAILED_TO_GENERATE_DRIVING_PERMIT_CLAIM);
-                }
-            }
+        var drivingPermitVc = findStoreItem(DRIVING_PERMIT_CRI_TYPES, successfulVCStoreItems);
+
+        if (drivingPermitVc.isEmpty()) {
+            LOGGER.warn("Failed to find Driving Permit CRI credential");
+            return Optional.empty();
+        }
+
+        var drivingPermitNode =
+                extractCriNodeFromCredential(
+                        DRIVING_PERMIT_PROPERTY_NAME,
+                        drivingPermitVc.get(),
+                        "Error while parsing Driving Permit CRI credential",
+                        ErrorResponse.FAILED_TO_GENERATE_DRIVING_PERMIT_CLAIM);
+
+        if (drivingPermitNode.isMissingNode()) {
+            StringMapMessage mapMessage =
+                    new StringMapMessage()
+                            .with(
+                                    LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                    "Driving Permit property is missing from VC")
+                            .with(
+                                    LOG_CRI_ISSUER.getFieldName(),
+                                    drivingPermitVc.get().getCredentialIssuer());
+            LOGGER.warn(mapMessage);
+
+            return Optional.empty();
+        }
+
+        if (drivingPermitNode instanceof ArrayNode) {
+            ((ObjectNode) drivingPermitNode.get(0)).remove("fullAddress");
+            ((ObjectNode) drivingPermitNode.get(0)).remove("issueDate");
+        }
+
+        return Optional.of(drivingPermitNode);
+    }
+
+    private Optional<VcStoreItem> findStoreItem(String criName, List<VcStoreItem> vcStoreItems) {
+        return vcStoreItems.stream()
+                .filter(credential -> credential.getCredentialIssuer().equals(criName))
+                .findFirst();
+    }
+
+    private Optional<VcStoreItem> findStoreItem(
+            List<String> criNames, List<VcStoreItem> vcStoreItems) {
+        return vcStoreItems.stream()
+                .filter(credential -> criNames.contains(credential.getCredentialIssuer()))
+                .findFirst();
+    }
+
+    private JsonNode extractCriNodeFromCredential(
+            String criName,
```

Comment:
Sorry, another small thing - this parameter isn't actually the CRI name, it's the property you're extracting. Perhaps `extractSubjectClaim()` or similar?